apis/kubernetes/v1alpha1/types.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// A ProviderSpec defines the desired state of a Provider.
+type ProviderSpec struct {
+	// Kubernetes credentials Secret reference
+	// A Secret containing connection credentials for a Kubernetes cluster
+	// client that will be used to authenticate to this Kubernetes Provider.
```

Comment:
This comment will be used to generate user facing API documentation, so I think there would be value in explaining what we expect this secret to look like. We probably don't need to go as far as outlining all the expected fields, but it would probably be useful to say something like "this will typically be the connection secret of a KubernetesCluster claim, or the secret created by a Kubernetes service account".

Reply:
Description fixed and docs generated using crossdocs!